ui/src/utils/state.js

```diff
@@ -1,68 +1,115 @@
+            icon: "check-circle-outline",
+            isRunning: false,
+            isKillable: false
+        },
+        RUNNING: {
+            name: "RUNNING",
+            colorClass: "primary",
+            color: "#1AA5DE",
+            icon: "play-circle-outline",
+            isRunning: true,
+            isKillable: true
+        },
+        KILLING: {
+            name: "KILLING",
+            colorClass: "warning",
+            color: "#FBD10B",
+            icon: "close-circle-outline",
+            isRunning: true,
+            isKillable: false
+        },
+        KILLED: {
+            name: "KILLED",
+            colorClass: "warning",
+            color: "#FBD10B",
+            icon: "stop-circle-outline",
+            isRunning: false,
+            isKillable: false
+        },
+        FAILED: {
+            name: "FAILED",
+            colorClass: "danger",
+            color: "#F04124",
+            icon: "close-circle-outline",
+            isRunning: false,
+            isKillable: false
+        }
+    });
+
     static get CREATED() {
-        return "CREATED";
+        return this.States.CREATED.name;
     }
 
     static get RESTARTED() {
-        return "RESTARTED";
+        return this.States.RESTARTED.name;
     }
 
     static get SUCCESS() {
-        return "SUCCESS";
+        return this.States.SUCCESS.name;
     }
 
     static get RUNNING() {
-        return "RUNNING";
+        return this.States.RUNNING.name;
     }
 
     static get KILLING() {
-        return "KILLING";
+        return this.States.KILLING.name;
     }
 
     static get KILLED() {
-        return "KILLED";
+        return this.States.KILLED.name;
     }
 
     static get FAILED() {
-        return "FAILED";
+        return this.States.FAILED.name;
     }
 
-    static colorClass() {
-        return {
-            [State.CREATED]: "info",
-            [State.RESTARTED]: "info",
-            [State.SUCCESS]: "success ",
-            [State.RUNNING]: "primary",
-            [State.KILLING]: "warning",
-            [State.KILLED]: "warning",
-            [State.FAILED]: "danger"
-        };
+    static isRunning(state) {
+        return this.States[state] && this.States[state].isRunning;
     }
 
-    static color() {
-        return {
-            [State.CREATED]: "#75bcdd",
-            [State.RESTARTED]: "#75bcdd",
-            [State.SUCCESS]: "#43ac6a",
-            [State.RUNNING]: "#1AA5DE",
-            [State.KILLING]: "#FBD10B",
-            [State.KILLED]: "#FBD10B",
-            [State.FAILED]: "#F04124"
-        };
+    static isKillable(state) {
+        return this.States[state] && this.States[state].isKillable;
+    }
+
+    static allStates() {
+        return _map(this.States, "name");
     }
 
-    static isRunning(current) {
-        return [State.KILLING, State.CREATED, State.RUNNING].includes(current);
+    static colorClass() {
+        return _mapValues(this.States, state => state.colorClass);
```

Comment:
I am not exactly sure about what does exactl _mapValues, but it looks like you want here a simple Array.map.
So you might use es6 core features like this for the same results:
```javascript
this.States.map(states => state.colorClass)
```

Reply:
this.States is not an array but object (pseudo map) ... so there's no map function on it ...
Didn't want to make this.States an array ... but a "pseudo" enum object ... (autocompletion in IDE ...)

Reply:
ok, no problems :+1:  All is good then.